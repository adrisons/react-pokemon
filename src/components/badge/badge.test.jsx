import React from "react";
import renderer from "react-test-renderer";
import typeColors from "../../helpers/type-colors";
import Badge from "./badge";
describe("WHEN: Badge", () => {
  let tree;
  beforeEach(() => {
    const component = renderer.create(<Badge name="bug" />);
    tree = component.toJSON();
  });
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
  it("THEN: should display correct border color", () => {
    expect(tree.props.style.border.includes(typeColors["bug"])).toBeTruthy();
    expect(tree.props.style.border.includes(typeColors["dragon"])).toBeFalsy();
  });
});
