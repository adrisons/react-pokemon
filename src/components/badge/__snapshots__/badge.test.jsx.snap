// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN: Badge THEN: should match snapshot 1`] = `
<div
  className="badge"
  style={
    Object {
      "border": "1px solid #729f3f",
      "borderLeft": "9px solid #729f3f",
    }
  }
>
  bug
</div>
`;
